moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_joints.cpp

```diff
@@ -0,0 +1,535 @@
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Robert Haschke */
+
+#include <moveit/motion_planning_rviz_plugin/motion_planning_frame_joints.h>
+#include <moveit/motion_planning_rviz_plugin/motion_planning_display.h>
+
+#include "ui_motion_planning_rviz_plugin_frame_joints.h"
+#include <QPainter>
+#include <QSlider>
+
+namespace moveit_rviz_plugin
+{
+JMGItemModel::JMGItemModel(const moveit::core::RobotState& robot_state, const std::string group_name, QObject* parent)
+  : QAbstractTableModel(parent), robot_state_(robot_state), jmg_(nullptr)
+{
+  if (robot_state_.getRobotModel()->hasJointModelGroup(group_name))
+    jmg_ = robot_state_.getRobotModel()->getJointModelGroup(group_name);
+}
+
+int JMGItemModel::rowCount(const QModelIndex& parent) const
+{
+  if (!jmg_)
+    return robot_state_.getVariableCount();
+  else
+    return jmg_->getVariableCount();
+}
+
+int JMGItemModel::columnCount(const QModelIndex& parent) const
+{
+  return 2;
+}
+
+Qt::ItemFlags JMGItemModel::flags(const QModelIndex& index) const
+{
+  if (!index.isValid())
+    return 0;
+
+  Qt::ItemFlags f = QAbstractTableModel::flags(index);
+  if (index.column() == 1)
+  {
+    const moveit::core::JointModel* jm = getJointModel(index);
+    if (!jm->isPassive() && !jm->getMimic())  // these are not editable
+      f |= Qt::ItemIsEditable;
+  }
+  return f;
+}
+
+QVariant JMGItemModel::data(const QModelIndex& index, int role) const
+{
+  if (!index.isValid())
+    return QVariant();
+
+  int idx = jmg_ ? jmg_->getVariableIndexList()[index.row()] : index.row();
+  switch (index.column())
+  {
+    case 0:  // joint name column
+      switch (role)
+      {
+        case Qt::DisplayRole:
+          return QString::fromStdString(robot_state_.getVariableNames()[idx]);
+        case Qt::TextAlignmentRole:
+          return Qt::AlignLeft;
+      }
+      break;
+    case 1:  // joint value column
+    {
+      double value = robot_state_.getVariablePosition(idx);
+      switch (role)
+      {
+        case Qt::DisplayRole:
+          return value;
+        case Qt::EditRole:
+          if (const moveit::core::JointModel* jm = robot_state_.getRobotModel()->getJointOfVariable(idx))
+            return jm->getType() == moveit::core::JointModel::REVOLUTE ? value * 180 / M_PI : value;
+          break;
+        case ProgressBarDelegate::JointTypeRole:
+          if (const moveit::core::JointModel* jm = robot_state_.getRobotModel()->getJointOfVariable(idx))
+            return jm->getType();
+          break;
+        case ProgressBarDelegate::PercentageRole:
+          if (const moveit::core::VariableBounds* bounds = getVariableBounds(index))
+            return static_cast<int>((value - bounds->min_position_) / (bounds->max_position_ - bounds->min_position_) *
```

Comment:
Does this correctly handle continuous joints?

Reply:
Didn't yet worked. Should be fixed now.